Remove PyCrypto

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Remove `PyCrypto` from Ansible since it is no longer maintained and not compatible with Python 3.10.

Fixes #74626.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`packaging/debian/rules`
`setup.py`
`test/units/parsing/vault`
`lib/ansible/executor/process/worker.py`
`lib/ansible/parsing/vault/__init__.py`
`docs/docsite/rst/installation_guide/intro_installation.rst`

samdoran committed:
Remove PyCrypto from setup.py and packaging script

samdoran committed:
Remove mention of pycrpto from installation docs

samdoran committed:
Remove unused import

samdoran committed:
Remove atfork() use in worker

samdoran committed:
Remove PyCrypto from vault

samdoran committed:
Remove pycryto constraint and unit test requirement

samdoran committed:
Remove integration test script that was not used

samdoran committed:
Remove PyCrypto tests from unit tests

ci_complete

samdoran committed:
Add docs and fix warning message

relrod commented on lib/ansible/parsing/vault/__init__.py:
```diff
@@ -1310,11 +1237,9 @@ def _decrypt_cryptography(cls, b_ciphertext, b_crypted_hmac, b_key1, b_key2, b_i
 
-        It would be nice if there was a library for this but
-        hey.
+        It would be nice if there was a library for this but hey.
```
```suggestion
        It would be nice if there were a library for this but hey.
```

samdoran committed:
Update comment

Co-authored-by: Rick Elrod <rick@elrod.me>

acozine commented:
Currently we list `cryptography` as an optional add-on for Vault, a way to speed it up: https://docs.ansible.com/ansible/latest/user_guide/vault.html#speeding-up-ansible-vault

Is it now a pre-requisite for using Vault? If so, we should update that page.

nitzmahone commented:
`cryptography` has been a required dependency for Ansible itself for awhile now, so we can probably just get rid of any verbiage about it at all in the vault docs...

samdoran committed:
Remove section about cryptography library in Ansible Vault docs

Ansible itself requires cryptography so this section about installing it for
Ansible Vault is unnecessary.

samdoran commented:
I removed the section in the Ansible Vault docs about requiring `cryptography`.

felixfontein commented:
@samdoran the issue mentioned in the changelog fragment is #72646 (which is unrelated to PyCrypto), not #74626 as mentioned in this PR.

felixfontein commented on docs/docsite/rst/porting_guides/porting_guide_core_2.12.rst:
```diff
@@ -25,7 +25,7 @@ No notable changes
 ============
 
-No notable changes
+* ``ansible-vault`` no longer supports ``PyCrypto`` and requires ``cryptograhpy``.
```
Typo: `cryptography`

samdoran replied:
🤦‍♂️
